### PR TITLE
Mcg monitoring

### PR DESCRIPTION
**Pull request by naveenpaul1**
Monitoring stack changes and ODF removed from the dependency

@v-harihar @aspandey

### CODE REVIEW

File: controllers/managedmcg_controller.go [hunk 1]
```diff
@@ -72,61 +69,68 @@ type ManagedMCGReconciler struct {
...
+	onboardingValidationKeySecret *corev1.Secret
+	alertmanager                  *promv1.Alertmanager
+	PagerdutySecretName           string
+	dmsRule                       *promv1.PrometheusRule
+	DeadMansSnitchSecretName      string
+	CustomerNotificationHTMLPath  string
+	SMTPSecretName                string
+	SOPEndpoint                   string
+	AlertSMTPFrom                 string
+	Route                         *routev1.Route
```
Comment (rexagod) [inline controllers/managedmcg_controller.go:97]: Please separate the exported and non-exported fields with one newline.

File: controllers/managedmcg_controller.go [hunk 2]
```diff
@@ -72,61 +69,68 @@ type ManagedMCGReconciler struct {
...
-	r.storageSystem.Name = storageSystemName
-	r.storageSystem.Namespace = r.namespace
-
-	r.operatorConsole = &operatorv1.Console{}
-	r.operatorConsole.Name = operatorConsoleName
-
-	r.storageCluster = &ocsv1.StorageCluster{}
-	r.storageCluster.Name = storageClusterName
-	r.storageCluster.Namespace = r.namespace
+	r.initPrometheusReconciler(req)
```
Comment (rexagod) [inline controllers/managedmcg_controller.go:112]:
```suggestion
	r.initializePrometheusReconciler(req)
```

File: controllers/managedmcg_controller.go [hunk 3]
```diff
@@ -72,61 +69,68 @@ type ManagedMCGReconciler struct {
...
+// +kubebuilder:rbac:groups="monitoring.coreos.com",namespace=system,resources=podmonitors,verbs=get;list;watch;update;patch
+// +kubebuilder:rbac:groups="monitoring.coreos.com",namespace=system,resources=servicemonitors,verbs=get;list;watch;update;patch;create
+// +kubebuilder:rbac:groups="route.openshift.io",namespace=system,resources=routes,verbs=get;list;watch;create;update
+// +kubebuilder:rbac:groups="apps",namespace=system,resources=statefulsets,verbs=get;list;watch
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
 func (r *ManagedMCGReconciler) Reconcile(_ context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := r.Log.WithValues("req.Namespace", req.Namespace, "req.Name", req.Name)
+	log := r.Log.WithValues("request_namespace", req.Namespace, "request_name", req.Name)
```
Comment (rexagod) [inline controllers/managedmcg_controller.go:133]: Why this change?

File: controllers/managedmcg_controller.go [hunk 4]
```diff
@@ -156,32 +160,23 @@ func (r *ManagedMCGReconciler) Reconcile(_ context.Context, req ctrl.Request) (c
 
 func (r *ManagedMCGReconciler) reconcilePhases() (reconcile.Result, error) {
 	r.Log.Info("reconciliation phases initiated")
-	foundAddonDeletionKey := r.verifyAddonDeletionKey() // TODO
-	r.updateNoobaaComponentStatus()
+	foundAddonDeletionKey := r.verifyAddonDeletionKey()
+	r.updateComponentStatus()
```
Comment (rexagod) [inline controllers/managedmcg_controller.go:164]: I thought we agreed on the previous name when the refactor PR went in?

File: controllers/managedmcg_controller.go [hunk 5]
```diff
@@ -531,3 +510,50 @@ func (r *ManagedMCGReconciler) lookupImages() error {
...
+		if strings.HasPrefix(candidate.Name, name) {
+			csv = &candidate
+			break
+		}
+	}
+	if csv == nil {
+		return nil, fmt.Errorf("unable to get csv resources for %s ", name)
+	}
+	return csv, nil
+}
```
Comment (rexagod) [inline controllers/managedmcg_controller.go:559]: Can we move these helper functions to `/utils`?

File: templates/route.go [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2022.
+
```
Comment (rexagod) [inline templates/route.go:2]: Why this change?

**Review by rexagod — CHANGES_REQUESTED** [review]

File: config/manager/kustomization.yaml [hunk 7]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: quay.io/red-hat-storage/mcg-osd-deployer
-  newTag: latest
+  newName: quay.io/osd-addons/mcg-osd
+  newTag: mcg-osd-deployer
```
[the remark this answers is not in the document]
Comment (naveenpaul1) [reply config/manager/kustomization.yaml:16]: yes, Tag should be something that points to the version of the image, we keep it `mcg-osd-deployer` only for now and once we have prod ready version we should change the version to a valid one (eg:- 1.0.0)

File: config/manager/manager.yaml [hunk 8]
```diff
@@ -26,7 +28,7 @@ spec:
         - name: readiness-server
           command:
             - /readinessServer
-          image: quay.io/red-hat-storage/mcg-osd-deployer:latest
+          image: controller:latest
```
[the remark this answers is not in the document]
Comment (naveenpaul1) [reply config/manager/manager.yaml:31]: even though its `controller:latest` is replaced with a correct image tag from another config file, quay.io/osd-addons/mcg-osd:mcg-osd-deployer

File: controllers/managedmcg_controller.go (hunk 1 above)
Comment (naveenpaul1) [reply controllers/managedmcg_controller.go:97]: updated

File: bundle.Dockerfile [hunk 9]
```diff
@@ -6,7 +6,7 @@ LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=mcg-osd-deployer
 LABEL operators.operatorframework.io.bundle.channels.v1=alpha
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.18.1
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.18.0+git
```
Comment (oritwas) [inline bundle.Dockerfile:9]: Why do we need to downgrade the operator sdk version?

File: controllers/managedmcg_controller.go (hunk 4 above)
Comment (naveenpaul1) [reply controllers/managedmcg_controller.go:164]: need to update status for Prometheus and AlerManager along with Noobaa, because of that method name `updateNoobaaComponentStatus()` dosnt make sense.

File: controllers/managedmcg_controller.go (hunk 5 above)
Comment (naveenpaul1) [reply controllers/managedmcg_controller.go:559]: now the util calss is out side the `controller` package because of that we cannot access `ManagedMCGReconciler` in util package

File: config/manager/kustomization.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (rexagod) [reply config/manager/kustomization.yaml:16]: Okay, in that case, since we have a non-prod-ready alpha version on our hands, can we point this to say, `0.1.0` (or any other version that we can agree upon based on the number of changes and features hat have went in) to indicate that?

File: controllers/managedmcg_controller.go (hunk 4 above)
Comment (rexagod) [reply controllers/managedmcg_controller.go:162]: Okay.

File: controllers/managedmcg_controller.go (hunk 5 above)
Comment (rexagod) [reply controllers/managedmcg_controller.go:557]: Can we send that as a function parameter instead of a receiver? Following this pattern will allow us to trim down help function population in the core logic.

File: bundle.Dockerfile (hunk 9 above)
Comment (naveenpaul1) [reply bundle.Dockerfile:9]: will update the version to latest version

File: config/manager/kustomization.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (naveenpaul1) [reply config/manager/kustomization.yaml:16]: Sure we can do that, will update the image tag

**Review by rexagod — APPROVED** [review]
LGTM, assuming we don't need routes anymore.
